ssl-config/src/main/java/com/palantir/remoting/ssl/KeyStores.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting.ssl;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Throwables;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+
+final class KeyStores {
+
+    private KeyStores() {
+    }
+
+    /**
+     * Returns a {@link KeyStore} created by loading the certificate or certificates specified by the provided path.
+     *
+     * @param path a path to a X.509 certificate in PEM or DER format, or to a directory containing certificate files.
+     * If the path specifies a directory, every non-hidden file in the directory must be a X.509 certificate in PEM or
+     * DER format.
+     * @return a new KeyStore of type {@link KeyStore#getDefaultType()} that contains the certificates specified by the
+     * provided path. The returned store will not have any password.
+     */
+    static KeyStore createTrustStoreFromCertificates(Path path) {
+        try {
+            KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            keyStore.load(null, null);
+
+            File pathFile = path.toFile();
+            if (pathFile.isDirectory()) {
+                File[] files = pathFile.listFiles(VISIBLE_FILE_FILTER);
```

Comment:
Don't think it should matter -- they're named entries, and fairly sure that for trust stores the provider will iterate through all of the certificates in the store when doing validation